client/lib/wp/sync-handler/index.js

```diff
@@ -187,13 +193,48 @@ export class SyncHandler {
 		return localforage.getItem( key, fn );
 	}
 
+	/**
+	 * Add/Override a record.
+	 *
+	 * @param {String} key - record key identifier
+	 * @param {Object} data - data to store
+	 * @param {Function} [fn] - callback function
+	 * @return {Promise} natuve ES6 promise
+	 */
 	storeRecord( key, data, fn = () => {} ) {
 		debug( 'storing data in %o key\n', key );
-		return localforage.setItem( key, data, fn );
+
+		// add this record to history
+		return cacheInvalidation
```

Comment:
I feel like this is so closely coupled that we can let the cache also set the record here, rather than doing it in a callback below.